app/src/main/java/com/example/quizacademytask/App.kt

```diff
@@ -0,0 +1,19 @@
+package com.example.quizacademytask
+
+import android.app.Application
+import android.content.Context
+import db.AppDatabase
+
+class App : Application() {
+
+    override fun onCreate() {
+        super.onCreate()
+        db = AppDatabase.getInstance(this)
+        context = applicationContext
+    }
+
+    companion object {
+        lateinit var db: AppDatabase
+        lateinit var context: Context
```

Comment:
there is no need to hold reference to application context here. Application already has getApplicationContext for it: https://developer.android.com/reference/android/content/Context#getApplicationContext()